src/main/java/edu/uw/edm/edmzuulproxy/filter/AcsUserSyncFilter.java

```diff
@@ -0,0 +1,53 @@
+package edu.uw.edm.edmzuulproxy.filter;
+
+import com.netflix.zuul.ZuulFilter;
+import com.netflix.zuul.context.RequestContext;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.security.Principal;
+
+import javax.servlet.http.HttpServletRequest;
+
+import edu.uw.edm.edmzuulproxy.service.UserProvisioningService;
+import lombok.extern.slf4j.Slf4j;
+
+import static org.springframework.cloud.netflix.zuul.filters.support.FilterConstants.ROUTE_TYPE;
+
+@Slf4j
+@Component
+public class AcsUserSyncFilter extends ZuulFilter {
+    private UserProvisioningService userProvisioningService;
+
+    @Autowired
+    public AcsUserSyncFilter(UserProvisioningService userProvisioningService) {
+        this.userProvisioningService = userProvisioningService;
+    }
+
+    @Override
+    public String filterType() {
+        return ROUTE_TYPE;
+    }
+
+    @Override
+    public int filterOrder() {
+        return 2;
+    }
+
+    @Override
+    public boolean shouldFilter() {
+        return true;
+    }
+
+    @Override
+    public Object run() {
+        RequestContext ctx = RequestContext.getCurrentContext();
+        HttpServletRequest request = ctx.getRequest();
+        Principal user = request.getUserPrincipal();
+
+        userProvisioningService.provisionAcsUser(user.getName());
```

Comment:
I'm wondering if we should `try/catch` the whole method to make sure that zuul never block request because of a user provisioning issue